contracts/UsingWitnet.sol

```diff
@@ -33,9 +33,10 @@ contract UsingWitnet {
   }
 
   // Ensures that user-specified rewards are equal to the total transaction value to prevent users from burning any excess value
-  modifier validRewards(uint256 _requestReward, uint256 _resultReward) {
-    require(_requestReward + _resultReward >= _requestReward, "The sum of rewards overflows");
-    require(msg.value == _requestReward + _resultReward, "Transaction value should equal the sum of rewards");
+  modifier validRewards(uint256 _requestReward, uint256 _resultReward, uint256 _blockReward) {
+    require(_requestReward + _resultReward + _blockReward >= _requestReward, "The sum of rewards overflows");
```

Comment:
This overflow check is wrong, with the values `_requestReward = 1`, `_resultReward = _blockReward = 2**256 - 1` it returns true:

    require(1 + (2**256 - 1) + (2**256 - 1) >= 1)

So use a different overflow check, and add tests.